File: sdks/php/src/Model/KeyVersion.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Model;

/**
 * A versioned public key for an entity.
 */
final class KeyVersion
{
    public function __construct(
        public readonly string $id,
        public readonly string $entityId = '',
        public readonly int $keyIndex = 0,
        public readonly string $validFrom = '',
        public readonly ?string $validUntil = null,
        public readonly string $createdAt = '',
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            entityId: (string) ($data['entity_id'] ?? ''),
            keyIndex: (int) ($data['key_index'] ?? 0),
            validFrom: (string) ($data['valid_from'] ?? ''),
            validUntil: $data['valid_until'] ?? null,
            createdAt: (string) ($data['created_at'] ?? ''),
        );
    }
}

File: sdks/java/src/main/java/dev/atap/sdk/model/KeyVersion.java
package dev.atap.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A versioned public key for an entity.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class KeyVersion {

    @JsonProperty("id")
    private String id;

    @JsonProperty("entity_id")
    private String entityId;

    @JsonProperty("key_index")
    private int keyIndex;

    @JsonProperty("valid_from")
    private String validFrom;

    @JsonProperty("valid_until")
    private String validUntil;

    @JsonProperty("created_at")
    private String createdAt;

    public KeyVersion() {
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getEntityId() { return entityId; }
    public void setEntityId(String entityId) { this.entityId = entityId; }

    public int getKeyIndex() { return keyIndex; }
    public void setKeyIndex(int keyIndex) { this.keyIndex = keyIndex; }

    public String getValidFrom() { return validFrom; }
    public void setValidFrom(String validFrom) { this.validFrom = validFrom; }

    public String getValidUntil() { return validUntil; }
    public void setValidUntil(String validUntil) { this.validUntil = validUntil; }

    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
}
